core/domain/topic_services.py

```diff
@@ -187,20 +198,50 @@ def _create_topic(committer_id, topic, commit_message, commit_cmds):
     generate_topic_summary(topic.id)
 
 
+def does_topic_with_name_exist(topic_name):
+    """Checks if the topic with provided name exists.
+
+    Args:
+        topic_name: str. The topic name.
+
+    Returns:
+        bool. Whether the the topic name exists.
+
+    Raises:
+        Exception. Topic name is not a string.
+    """
+    if not isinstance(topic_name, python_utils.BASESTRING):
+        raise utils.ValidationError('Name should be a string.')
+    existing_topic = topic_fetchers.get_topic_by_name(topic_name)
+    return existing_topic is not None
+
+
+def does_topic_with_url_fragment_exist(url_fragment):
+    """Checks if topic with provided url fragment exists.
+
+    Args:
+        url_fragment: str. The url fragment for the topic.
+
+    Returns:
+        bool. Whether the the url fragment for the topic exists.
+
+    Raises:
+        Exception. Topic URL fragment is not a string.
+    """
+    if not isinstance(url_fragment, python_utils.BASESTRING):
+        raise utils.ValidationError('Topic URL fragment should be a string.')
+    existing_topic = (
+        topic_fetchers.get_topic_by_url_fragment(url_fragment))
+    return existing_topic is not None
+
+
 def save_new_topic(committer_id, topic):
     """Saves a new topic.
 
     Args:
         committer_id: str. ID of the committer.
         topic: Topic. Topic to be saved.
-
-    Raises:
-        Exception. Topic with same name already exists.
     """
-    existing_topic = topic_fetchers.get_topic_by_name(topic.name)
```

Comment:
Why delete?

Reply:
This check can be done before calling save_new_topic. Like we already do in NewTopicHandler in the topics_and_skills_dashboard controller.